dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/DynamicPureLoopCutting.java

```diff
@@ -141,6 +149,11 @@ private void insertSideEffectChecks(IterationData iter) {
         assumeSideEffect.addFilters(Tag.SPINLOOP, Tag.EARLYTERMINATION, Tag.NOOPT);
         final Event spinloopStart = iterInfo.getIterationStart();
         assumeSideEffect.setCFileInformation(spinloopStart.getCLine(), spinloopStart.getSourceCodeFile());
+        // DynamicPureLoopCutting is run after unrolling, thus we need to assign all
+        // three of oId, cId and uId.
+        assumeSideEffect.setOId(spinloopStart.getOId());
+        assumeSideEffect.setCId(spinloopStart.getCId());
+        assumeSideEffect.setUId(spinloopStart.getUId());
```

Comment:
Same as above

Reply:
"Same as above" as "copy all meta-data" or "does not have any reasonable association with the metadata you attach to it"? If the later, I don't agree. We are even setting all the `CFileInformation` due to the association

Reply:
Both I guess. if you copy all ids and the source information, then you are copying everything, so just do `copyMetadata`.
The argument about already associating `CFileInformation` is only semi-valid.
First, one may argue that associating the `CFileInformation` is questionable to begin with. Indeed, I was surprised that we already did (cause I didn't add this when writing the code!), but I remember that you added it to track which loops ended up in deadlocks. This is a totally fine reason, and I agree there.
However, I don't think that matching `Cline` in any way correlates with the other ids also matching. I mean, it is totally common that two same-Cline events differ in all their ids.
In any case, matching this event with the loop back jump or the loop header is reasonable (as discussed above), so I'm fine with this.